### PR TITLE
[ODS-4808] Integrate Latest  3.3.0-a Data Standard Work (As Of 01/29/2021)

### PR DESCRIPTION
**Pull request by semalaiappan**

### CODE REVIEW

File: Artifacts/MsSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql [hunk 1]
```diff
@@ -0,0 +1,33 @@
+-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+DECLARE @applicationId INT;
```
Comment (gmcelhanon) [inline Artifacts/MsSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql:6]: Should there be a corresponding script file for Postgres at _Artifacts/PgSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql_ ? I don't see it in the PR.

**Review by gmcelhanon — CHANGES_REQUESTED** [review]

File: Artifacts/PgSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    THEN
+        SELECT ResourceClaimId INTO relationshipBasedDataResourceClaim_Id
+    	FROM dbo.ResourceClaims WHERE ResourceName = 'systemDescriptors' AND Application_ApplicationId = application_id;
+    END IF;
+
+   /* new descriptors */
+	IF (NOT EXISTS (SELECT 1 FROM dbo.ResourceClaims WHERE ResourceName = 'AncestryEthnicOriginDescriptor' AND Application_ApplicationId = application_id))
+	THEN
+	INSERT INTO dbo.ResourceClaims (DisplayName, ResourceName, ClaimName, ParentResourceClaimId, Application_ApplicationId)
+	VALUES (N'AncestryEthnicOriginDescriptor', N'AncestryEthnicOriginDescriptor', N'http://ed-fi.org/ods/identity/claims/AncestryEthnicOriginDescriptor', systemDescriptorsResourceClaim_Id, application_id);
```
Comment (gmcelhanon) [inline Artifacts/PgSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql:35]: The URI doesn't match conventions. The name of the resource in the URI should be camel-cased, as follows:
`http://ed-fi.org/ods/identity/claims/ancestryEthnicOriginDescriptor`
...not... `http://ed-fi.org/ods/identity/claims/AncestryEthnicOriginDescriptor`.

Ensure this convention is applied to all resource claims you're creating.

File: Artifacts/PgSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	END IF;
+	
+	-- Try insert StudentDisciplineIncidentBehaviorAssociation
+	IF (NOT EXISTS (SELECT 1 FROM dbo.ResourceClaims WHERE ResourceName = 'StudentDisciplineIncidentBehaviorAssociation' AND Application_ApplicationId = application_id))
+	THEN
+	INSERT INTO dbo.ResourceClaims (DisplayName, ResourceName, ClaimName, ParentResourceClaimId, Application_ApplicationId)
```
Comment (gmcelhanon) [inline Artifacts/PgSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql:41]: As a general comment on formatting... make sure that once you swap <kbd>TAB</kbd>s for spaces... that you properly indent these SQL statements with the `IF` block. The SQL should be indented for easier reading and maintenance.

File: Artifacts/MsSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+SELECT @applicationId = (SELECT applicationid FROM  dbo.Applications  WHERE  ApplicationName  = 'Ed-Fi ODS API');
+SELECT @systemDescriptorsResourceClaimId = (SELECT ResourceClaimId FROM dbo.ResourceClaims WHERE ResourceName = 'systemDescriptors' AND Application_ApplicationId = @applicationId);
+SELECT @relationshipBasedDataResourceClaimId = (SELECT ResourceClaimId FROM dbo.ResourceClaims WHERE ResourceName = 'relationshipBasedData' AND Application_ApplicationId = @applicationId);
+
+/* new descriptors */
+IF (NOT EXISTS (SELECT ResourceClaimId FROM dbo.ResourceClaims WHERE ResourceName = 'AncestryEthnicOriginDescriptor' AND Application_ApplicationId = @applicationId))
+BEGIN
+	INSERT dbo.ResourceClaims (DisplayName, ResourceName, ClaimName, ParentResourceClaimId, Application_ApplicationId)
+	VALUES (N'AncestryEthnicOriginDescriptor', N'AncestryEthnicOriginDescriptor', N'http://ed-fi.org/ods/identity/claims/AncestryEthnicOriginDescriptor', @systemDescriptorsResourceClaimId, @applicationId);
```
Comment (gmcelhanon) [inline Artifacts/MsSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql:18]: I left a similar comment on the Postgres version of this script, but you need to follow the conventions for the claim name URIs... which is to camel-case the resource name, as follows:
`http://ed-fi.org/ods/identity/claims/ancestryEthnicOriginDescriptor`
... not ... `http://ed-fi.org/ods/identity/claims/AncestryEthnicOriginDescriptor`.

Apply this convention to all resource claims you're creating.

File: Artifacts/MsSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+END
+
+-- Try insert StudentDisciplineIncidentBehaviorAssociation
+IF (NOT EXISTS (SELECT ResourceClaimId FROM dbo.ResourceClaims WHERE ResourceName = 'StudentDisciplineIncidentBehaviorAssociation' AND Application_ApplicationId = @applicationId))
+BEGIN
+INSERT [dbo].[ResourceClaims] ([DisplayName], [ResourceName], [ClaimName], [ParentResourceClaimId], [Application_ApplicationId])
```
Comment (gmcelhanon) [inline Artifacts/MsSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql:24]: See comment on Postgres script below regarding proper formatting and indenting.

**Review by gmcelhanon — DISMISSED** [review]

**Comment by gmcelhanon** [comment]
@semalaiappan You'll also need to bump the code gen package version as another CodeGen change went in today.

File: Artifacts/MsSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql (hunk 1 above)
Comment (semalaiappan) [reply Artifacts/MsSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql:6]: I added Artifacts/PgSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql

File: Artifacts/PgSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    relationshipBasedDataResourceClaim_Id INT;
+    
+BEGIN
+    IF  EXISTS (SELECT 1 FROM  dbo.Applications  WHERE  ApplicationName  = 'Ed-Fi ODS API')
+    THEN
+        SELECT applicationid INTO application_id
+    	FROM  dbo.Applications  WHERE  ApplicationName  = 'Ed-Fi ODS API';
+    END IF;
+
+ 	IF  EXISTS (SELECT 1 FROM dbo.ResourceClaims WHERE ResourceName = 'systemDescriptors' AND Application_ApplicationId = application_id)
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply Artifacts/PgSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql]: I updated . Thanks

File: Artifacts/PgSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql (hunk 2 above)
Comment (semalaiappan) [reply Artifacts/PgSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql]: OK sure .Thanks . I updated .

File: Artifacts/PgSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql (hunk 3 above)
Comment (semalaiappan) [reply Artifacts/PgSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql]: yes I removed tabs and used indent to do the formating now . Thanks

File: Artifacts/MsSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql (hunk 4 above)
Comment (semalaiappan) [reply Artifacts/MsSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql]: I fixed it . Thanks

File: Artifacts/MsSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql (hunk 5 above)
Comment (semalaiappan) [reply Artifacts/MsSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql]: yes ,I fixed it . Thanks

File: Artifacts/PgSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql (hunk 6 above)
[the remark this answers is not in the document]
Comment (semalaiappan) [reply Artifacts/PgSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql]: image
image

**Comment by semalaiappan** [comment]
> @semalaiappan You'll also need to bump the code gen package version as another CodeGen change went in today.

I took latest of main branch and rebase it with my branch . Then I created new build for codegen and updated  the package version

**Comment by semalaiappan** [comment]
@gmcelhanon,All review comments are fixed and ready for your final review now

**Review by blmeyers — APPROVED** [review]

**Pull request merged**